src/core/tests/Test_PatchInstaller.py

```diff
@@ -229,6 +230,50 @@ def test_apt_install_skips_esm_packages(self):
         obj.mock_unimport_uaclient_update_module()
         version_obj.mock_unimport_uaclient_version_module()
 
+    def test_patch_installer_for_azgps_coordinated(self):
+        argument_composer = ArgumentComposer()
+        argument_composer.maximum_duration = "PT235M"
+        argument_composer.health_store_id = "pub_offer_sku_2024.04.01"
+        runtime = RuntimeCompositor(argument_composer.get_composed_arguments(), True, Constants.APT)
+        runtime.package_manager.custom_sources_list = os.path.join(argument_composer.temp_folder, "temp2.list")
+        # Path change
+        runtime.set_legacy_test_type('HappyPath')
+        self.assertTrue(runtime.patch_installer.start_installation())
+        self.assertEqual(runtime.execution_config.max_patch_publish_date, "20240401T000000Z")
+        self.assertEqual(runtime.package_manager.max_patch_publish_date,"20240401T000000Z")  # supported and conditions met
+        runtime.stop()
+
+        argument_composer.maximum_duration = "PT30M"
+        runtime = RuntimeCompositor(argument_composer.get_composed_arguments(), True, Constants.APT)
+        runtime.set_legacy_test_type('HappyPath')
+        self.assertFalse(runtime.patch_installer.start_installation())                 # failure is in unrelated patch installation batch processing
+        self.assertEqual(runtime.execution_config.max_patch_publish_date, "20240401T000000Z")
+        self.assertEqual(runtime.package_manager.max_patch_publish_date, "")    # reason: not enough time to use
+        runtime.stop()
+
+        argument_composer.maximum_duration = "PT235M"
+        runtime = RuntimeCompositor(argument_composer.get_composed_arguments(), True, Constants.APT)
+        runtime.set_legacy_test_type('HappyPath')
+        runtime.package_manager.install_security_updates_azgps_coordinated = lambda: (1, "Failed")
+        self.assertFalse(runtime.patch_installer.start_installation())
+        self.assertEqual(runtime.execution_config.max_patch_publish_date, "20240401T000000Z")
+        self.assertEqual(runtime.package_manager.max_patch_publish_date, "")    # reason: not enough time to use
```

Comment:
Why is this due to not enough time?

Reply:
The reason should be fallback on default patching due to install_security_updates_azgps_coordinated failing right?

Reply:
Actual comment -- # reason: the strict SDP is forced to fail with the lambda above (updated)